public/cli/lib/commands/convert.ts

```diff
@@ -139,3 +143,23 @@ function getOutputFilePath(
     return path.join(outputDirectory, outputFilename);
   }
 }
+
+function checkOverwrite(outputFilePath: string, overwriteOutputFiles: boolean): string {
+  let filePath = outputFilePath;
+  // check if output file already exists:
+  if (fs.existsSync(filePath)) {
+    if (overwriteOutputFiles) {
+      // remove the existing file
+      fs.unlinkSync(filePath);
+    } else {
+      // get a list of all the files in the directory,
+      const listAllFiles = fs.readdirSync(path.dirname(outputFilePath), 'utf8');
+      // get a count of the ones that have a similar start name
+      const nameParts = path.parse(filePath);
+      const matchingFiles = (listAllFiles ?? []).filter((f) => f.startsWith(nameParts.name));
+      // since they want to keep the file then modify the filename - new filename = name (length).ext
+      filePath = path.join(nameParts.dir, `${nameParts.name} (${matchingFiles.length})${nameParts.ext}`);
```

Comment:
Using the number of similar files probably isn't the best way to rename the file. Creates some cases where the new file isn't saved.
e.g. folder contains: "Mii_Mark_1" and "Mii_Mark_1 (2)"
Maybe make the extension number fill in the gap: "Mii_Mark_1 (1)", or continue after the largest number: "Mii_Mark_1 (3)"

Reply:
This should be good now. I've added integration tests to prove some of the edge cases.

Reply:
There is still an edge case of a file named "Mii_Mark_1abcdefg (1)" or similar. Which will be sorted to the back of the list and then selected as the highest copy count. Maybe something like this might be better:
```javascript
const copyFilenamePattern = new RegExp("^" + nameParts.name + " \\((\\d+)\\)\\" + nameParts.ext + "$");
```
So that it checks the extension and nothing extra can come before, in between or after.